### ISSUE TITLE
rustfmt's eagerness to align array items with braces causes it to go over line limit

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang-nursery/rustfmt#1388]
Before:

```rust
// Paths that may contain platform-specific code
const EXCEPTION_PATHS: &'static [&'static str] = &[
    // std crates
    "src/liballoc_jemalloc",
    "src/liballoc_system",
    "src/libcompiler_builtins",
    "src/liblibc",
    "src/libpanic_abort",
    "src/libpanic_unwind",
    "src/libunwind",
    "src/libstd/sys/", // Platform-specific code for std lives here.
                       // This has the trailing slash so that sys_common is not excepted.
    "src/libstd/os", // Platform-specific public interfaces
    "src/rtstartup", // Not sure what to do about this. magic stuff for mingw

    // temporary exceptions
    "src/libstd/rtdeps.rs", // Until rustbuild replaces make
    "src/libstd/path.rs",
    "src/libstd/f32.rs",
    "src/libstd/f64.rs",
    "src/libstd/sys_common/mod.rs",
    "src/libstd/sys_common/net.rs",
    "src/libterm", // Not sure how to make this crate portable, but test needs it
    "src/libtest", // Probably should defer to unstable std::sys APIs

    // std testing crates, ok for now at least
    "src/libcoretest",

    // non-std crates
    "src/test",
    "src/tools",
    "src/librustc",
    "src/librustdoc",
    "src/libsyntax",
    "src/bootstrap",
];
```

After rustfmt:

```rust
// Paths that may contain platform-specific code
const EXCEPTION_PATHS: &'static [&'static str] = &[// std crates
                                                   "src/liballoc_jemalloc",
                                                   "src/liballoc_system",
                                                   "src/libcompiler_builtins",
                                                   "src/liblibc",
                                                   "src/libpanic_abort",
                                                   "src/libpanic_unwind",
                                                   "src/libunwind",
                                                   "src/libstd/sys/", // Platform-specific code for std lives here.
                                                   // This has the trailing slash so that sys_common is not excepted.
                                                   "src/libstd/os", // Platform-specific public interfaces
                                                   "src/rtstartup", // Not sure what to do about this. magic stuff for mingw

                                                   // temporary exceptions
                                                   "src/libstd/rtdeps.rs", // Until rustbuild replaces make
                                                   "src/libstd/path.rs",
                                                   "src/libstd/f32.rs",
                                                   "src/libstd/f64.rs",
                                                   "src/libstd/sys_common/mod.rs",
                                                   "src/libstd/sys_common/net.rs",
                                                   "src/libterm", // Not sure how to make this crate portable, but test needs it
                                                   "src/libtest", // Probably should defer to unstable std::sys APIs

                                                   // std testing crates, ok for now at least
                                                   "src/libcoretest",

                                                   // non-std crates
                                                   "src/test",
                                                   "src/tools",
                                                   "src/librustc",
                                                   "src/librustdoc",
                                                   "src/libsyntax",
                                                   "src/bootstrap"];
```

And then it complains:

```
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:61: line exceeded maximum length (maximum: 100, found: 115) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:62: line exceeded maximum length (maximum: 100, found: 117) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:63: line exceeded maximum length (maximum: 100, found: 106) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:64: line exceeded maximum length (maximum: 100, found: 124) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:67: line exceeded maximum length (maximum: 100, found: 107) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:73: line exceeded maximum length (maximum: 100, found: 128) (sorry)
Rustfmt failed at /Users/corey/dev/rust/src/tools/tidy/src/pal.rs:74: line exceeded maximum length (maximum: 100, found: 116) (sorry)
```

Is there any way to improve the experience here?

**Comment by topecongiro** [comment]
This is fixed on the current master (0.8.3 (c89c01e)).

**Comment by nrc** [comment]
Closed by #1502
